File: java/pl/edu/pg/po/projekt2/zwierzeta/Owca.java
package pl.edu.pg.po.projekt2.zwierzeta;

import pl.edu.pg.po.projekt2.Punkt;
import pl.edu.pg.po.projekt2.Swiat;
import pl.edu.pg.po.projekt2.Zwierze;

public class Owca extends Zwierze {
    private static final char symbol = 'O';
    private static final int inicjatywa = 4;

    public Owca(Swiat swiat, Punkt pole, int turaPowstania) {
        super(swiat, pole, turaPowstania, 4, inicjatywa, symbol);
        this.sila = 4;
        this.pole = pole;
        this.swiat = swiat;
        this.umarl = false;
        this.turaPowstania = turaPowstania;
    }

    @Override
    public String nazwaOrganizmu() {
        return "Owca";
    }
}

File: cpp/Owca.cpp
#include "Owca.h"
#include "Komentarz.h"

Owca::Owca(Swiat *swiat, Punkt pole, int turaPowstania)
    : Zwierze(swiat, pole, turaPowstania, sila, inicjatywa, symbol){
    this->symbol = SYMBOL_OWCY;
    this->sila = SILA_OWCY;
    this->inicjatywa = INICJATYWA_OWCY;
    this->pole = pole;
    this->swiat = swiat;
    this->umarl = false;
    this->turaPowstania = turaPowstania;
}

string Owca::nazwaOrganizmu() {
    return "Owca";
}
